File: spec/fixtures/knowledge/sample.ts
interface Greeter {
  greet(): string;
}

class Person extends BaseEntity {
  private name: string;

  constructor(name: string) {
    super();
    this.name = name;
  }

  greet(): string {
    return `Hello, ${this.name}`;
  }
}

class SimpleWorker {
  perform(): void {
    console.log("working");
  }
}

function createPerson(name: string, age: number): Person {
  return new Person(name);
}

type StringMap = Record<string, string>;

File: spec/fixtures/knowledge/sample.rb
# frozen_string_literal: true

module SampleModule
  class SampleClass
    def initialize(name)
      @name = name
    end

    def greet
      "Hello, #{@name}"
    end

    def farewell
      "Goodbye, #{@name}"
    end
  end

  class AnotherClass
    def initialize
      @started = true
    end

    def perform
      true
    end
  end
end
